enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,275 @@
+* Using roles to define reusable steps scoped to units of work
+* Use playbooks to set up common e2e workflows and chains for the project
+
+Thinking about a general pattern of how to organize the repository:
+
+```
+playbooks/
+  example-e2e-workflow.yml
+roles/
+    common/
+        tasks/
+        handlers/
+        library/
+        files/
+        templates/
+        vars/
+        defaults/
+        meta/
+    example_role/
+        tasks/
+        defaults/
+        meta/
+```
+
+This is pretty open ended and meant to be. I just believe that focusing on making
+reusable roles here for the steps is key.
+
+These are my thoughts about Openshift CI and the step registry and making it easy to understand. 
+Copy the `pre`, `test`, and `post` pattern locally in the playbooks and include the playbooks like so:
+
+```
+# File structure:
+playbooks/
+  example-e2e-workflow.yml
+  example-e2e-workflow_pre.yml
+  example-e2e-workflow_test.yml
+  example-e2e-workflow_post.yml
+
+# example-e2e-workflow.yml
+- hosts: localhost
+
+- name: Run e2e-workflow Pre steps
+  import_playbook: example-e2e-workflow_pre.yml
+
+- name: Run e2e-workflow Test steps
+  import_playbook: example-e2e-workflow_test.yml
+
+- name: Run e2e-workflow Post steps
+  import_playbook: example-e2e-workflow_test.yml
+```
+
+This is just a general idea and would like to keep this pretty open ended at the moment.
+The first priority will be to run these playbooks locally but how to understand this in the CI
+is being taken into consideration.
+
+### Open Questions [optional]
+
+### Test Plan
+
+* e2e tests for the three install scenarios of cluster 0 (SNO, Compact, HA)
+* Run Openshift compliance tests against these installed clusters
+* Unit tests for new code to reach 50% code coverage
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+* Ability to utilize the e2e framework locality to aid in development
+* Openshift compliance tests are run against the installed clusters
+* The e2e framework is on boarded into Openshift CI for the team's use
+* Sufficient test coverage (50% coverage for unit tests)
+
+#### Tech Preview -> GA
+
+* Add upgrade/downgrade e2e tests
+* Increase unit test code coverage to 75% or higher
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+Since the ephemeral agent based installer is targeted to be released with OCP it would 
+be easiest to version the e2e framework the same as OCP x.y. This is the strategy
+that openshift-tests takes.
+
+### Operational Aspects of API Extensions
+
+#### Failure Modes
+
+#### Support Procedures
+
+## Implementation History
+
+## Drawbacks
+
+The biggest argument against it is that by creating a new framework we don't
+inherit the work that was done previously in validating Assisted Installer.
+
+## Alternatives
+
+Alternatively we can re-factor one of the existing frameworks to include this
+new project.
```

Comment:
I would tend to agree on this point. Even though dev-scripts has some complexity, it seems covering most of the side tasks that may be required - for that maybe it could help trying to identify as much as possible the existing gaps